lemmy_apub/src/objects/community.rs

```diff
@@ -65,15 +65,15 @@ impl ToApub for Community {
       group.set_content(d);
     }
 
-    if let Some(icon) = &self.icon {
+    if let Some(icon_url) = &self.icon {
       let mut image = Image::new();
-      image.set_url(icon.to_owned());
+      image.set_url(Url::parse(icon_url)?);
```

Comment:
Does this really make any difference?

Reply:
Maybe not, but it seems safe to do this check. I grepped for all instances of `set_url` and found these ones that didn't do the check.